docs/development/patches.md

```diff
@@ -0,0 +1,84 @@
+
+> Side note: the reason we use a `.patches` file to maintain the order of applied patches, rather than prepending a number like `001-` to each file, is because it reduces conflicts related to patch ordering. It prevents the situation where two PRs both add a patch at the end of the series with the same numbering and end up both getting merged resulting in a duplicate identifier, and it also reduces churn when a patch is added or deleted in the middle of the series.
+
+### Usage
+
+#### Adding a new patch
+```
+$ cd src/third_party/electron_node
+$ vim some/code/file.cc
+$ git commit
+$ ../../electron/script/git-export-patches -o ../../electron/patches/node
+```
+
+Re-exporting patches will sometimes cause shasums in unrelated patches to change. This is generally harmless and can be ignored (but go ahead and add those changes to your PR, it'll stop them from showing up for other people).
+
+#### Editing an existing patch
+```
+$ cd src/v8
+$ vim some/code/file.cc
+$ git log
+# Find the commit sha of the patch you want to edit.
+$ git commit --fixup [COMMIT_SHA]
+$ git rebase --autosquash -i [COMMIT_SHA]^
+$ ../electron/script/git-export-patches -o ../electron/patches/v8
+```
+
+#### Removing a patch
+```
+$ vim src/electron/patches/node/.patches
+# Delete the line with the name of the patch you want to remove
+$ cd src/third_party/electron_node
+$ git reset --hard refs/patches/upstream-head
+$ ../../electron/script/git-import-patches ../../electron/patches/node
+$ ../../electron/script/git-export-patches -o ../../electron/patches/node
+```
+
+Note that `git-import-patches` will mark the commit that was `HEAD` when it was run as `refs/patches/upstream-head`. This lets you keep track of which commits are from Electron patches (those that come after `refs/patches/upstream-head`) and which commits are in upstream (those before `refs/patches/upstream-head`).
+
+#### Resolving conflicts
+When updating an upstream dependency, patches may fail to apply cleanly. Often, the conflict can be resolved automatically by git with a 3-way merge. You can instruct `git-import-patches` to use the 3-way merge algorithm by passing the `-3` argument:
+
+```
+$ cd src/third_party/electron_node
+$ ../../electron/script/git-import-patches -3 ../../electron/patches/node
+```
+
+If `git-import-patches -3` encounters a merge conflict that it can't resolve automatically, it will pause and allow you to resolve the conflict manually. Once you have resolved the conflict, `git add` the resolved files and continue to apply the rest of the patches by running `git am --continue`.
```

Comment:
In the upgrade hackmd we say to run:
 `git am --abort`, then `git checkout refs/patches/upstream-head` before running the 3 way merge.  Is that necessary and should it be listed here?